lib/jmap/core/email/filter.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:equatable/equatable.dart';
+import 'package:jmap_dart_client/http/converter/mailbox_id_converter.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'filter.g.dart';
+
+@MailboxIdConverter()
+@JsonSerializable()
+class Filter with EquatableMixin {
+
+  @JsonKey(name: 'inMailbox', includeIfNull: false)
```

Comment:
no, it should not be in the core. `core` should not be care about mailbox or detail thing. Specs describe it carefully, we need to follow that to make our library can work with not only Tmail, but also other Jmap server